src/dfx/src/lib/identity/identity_manager.rs

```diff
@@ -642,17 +635,20 @@ fn remove_identity_file(file: &Path) -> DfxResult {
 }
 
 /// Generates a new secp256k1 key.
-#[context("Failed to generate a fresh secp256k1 key.")]
-pub(super) fn generate_key() -> DfxResult<(Vec<u8>, Mnemonic)> {
-    let mnemonic = Mnemonic::new(MnemonicType::for_key_size(256)?, Language::English);
+pub(super) fn generate_key() -> Result<(Vec<u8>, Mnemonic), IdentityError> {
+    let mnemonic = Mnemonic::new(MnemonicType::for_key_size(256).unwrap(), Language::English);
```

Comment:
This `.unwrap()` call is on a constant, so I didn't think it was worth creating an error type/description for something that can never happen.